addons/mail/static/src/core/common/message_model.js

```diff
@@ -365,6 +365,47 @@ export class Message extends Record {
         );
     }
 
+    get messagePreviewText() {
+        if (!(this.isBodyEmpty && this.attachment_ids.length) || this.subtype_description) {
+            return this.inlineBody || this.subtype_description;
+        }
+        const { attachment_ids: attachments } = this;
+        if (!attachments || attachments.length === 0) return "";
+
+        const attachmentName =
+            attachments[0].mimetype === "audio/mpeg" && attachments[0].voice
+                ? _t("Voice Message")
+                : attachments[0].name || "";
+
+        switch (attachments.length) {
+            case 1:
+                return attachmentName;
+            case 2:
+                return _t("%(attachmentname)s and %(count)s other attachment.", {
+                    attachmentname: attachmentName,
+                    count: attachments.length - 1,
+                });
+            default:
+                return _t("%(attachmentname)s and %(count)s other attachments.", {
+                    attachmentname: attachmentName,
+                    count: attachments.length - 1,
+                });
```

Comment:
The case for 2 and more are currently having the exact same text?
It's supposed to be "attachment 1" and "attachment 2" for the "2" case rather than displaying one name and then count.